### ISSUE TITLE
Unable to commit on Windows with 2.23.0

### ISSUE THREAD

**Issue by mkleehammer** [magit/magit#3955]
After upgrading git, with no magit upgrade, only empty commits were created.  This was true if I had no staged files and let the commit command stage all or if I staged them first and attempted to commit.  I also tried staging files and amending the empty commit and that did not work either.  In all cases, the commit was empty and the files were shown as not staged. 

When I downgraded git on 2.22, it worked again.

Unfortunately I didn't test git on the command-line before downgrading as I was troubleshooting and didn't know what I'd need and I don't have time to do so today.  So I don't really know if it is a git on Windows error, a magit staging problem with 2.23, or a magit commit problem with 2.23.  I can return to it and test later if you don't have a Windows VM.

Since the upgrade to git was prompted by the daily check, I thought I'd give you a heads up since you might see a lot of them.  If you need to refer someone to the 2.22 installer: https://github.com/git-for-windows/git/releases/tag/v2.22.0.windows.1 

----

There are no errors and the git process buffer had the same contents it always does.

----

Magit 20190825.1416, Git 2.22.0.windows.1, Emacs 26.2, windows-nt

I did upgrade magit during by troubleshooting, so I don't know the earlier version that wasn't working, but the latest did not either.  This is installed from MELPA using Spacemacs.

**Comment by tarsius** [comment]
I suspect this isn't a Magit issue.

I recommend you first try to reproduce the issue on the command line. If `git commit -m bla` works, then progress to EDITOR=nano git commit`.

**Comment by tarsius** [comment]
Also note that `git commit` refuses to create empty commits without `--allow-empty`. If it does so anyway, then there is definitely something wrong with `git` itself.

I am closing for now. Please let us know if there is anything that indicates that Magit contributes to the issue.

**Comment by npostavs** [comment]
FYI, I just tried updating to `Git 2.23.0.windows.1` and magit seems to commit with no problems.

**Comment by mkleehammer** [comment]
Something else is up.  I rolled back to 2.22 on Windows and it started working.  Today, however, I started getting the same error on macOS with 2.23.0.  However, I hadn't changed anything in at least a week or two.

I was able to commit on the command line using `git add -a` and `git commit -m "xxx"`.

This was in the magit-process buffer:

```
  0 git … add -u .
  0 git … commit --
error: pathspec ':/' did not match any file(s) known to git
hint: Waiting for your editor to close the file...
Waiting for Emacs...
[<branch> <commit>] <msg>
```

I masked the items in the last line, but they were the correct branch, commit, and message.

The result was an empty commit.  The commit exists, but has no files.
